core/kazoo_services/src/kz_services.erl

```diff
@@ -1184,6 +1185,92 @@ save_services_jobj(Services, ProposedJObj) ->
+    AccountId = account_id(Services),
+    ResellerId = kz_services_reseller:get_id(AccountId),
+
+    case kz_services_reseller:is_reseller(AccountId)
+         andalso MasterAccountId =/= AccountId of
+        % don't cascade audit to Master Acct
+        'false' -> 'ok';
+        'true' ->
+            lager:debug("creating account ~s reseller audit for ~s", [AccountId, ResellerId]),
+            JObj = kz_json:from_list_recursive([{<<"id">>, kazoo_yodb_util:yodb_id()}
+                                               ,{<<"audit">>
+                                                ,generate_diff_obj(CurrentServicesJObj, ProposedServicesJObj)
+                                                }
+                                               ]),
+            Doc = kz_doc:update_pvt_parameters(JObj, kazoo_yodb:get_yodb(AccountId)),
+            _ = kazoo_yodb:save_doc(AccountId, Doc),
+            'ok'
+    end.
+
+%%------------------------------------------------------------------------------
+%% @doc run a diff between services jobs to determine if cascade quantities
+%% changed.
+%% @end
+%%------------------------------------------------------------------------------
+-spec has_quantity_changes(kz_json:object(), kz_json:object()) -> boolean().
+has_quantity_changes(CurrentServicesJObj, ProposedServicesJObj) ->
+    CascadesQty = [<<"quantities">>],
+    PQ = kz_json:get_value(CascadesQty, ProposedServicesJObj),
+    CQ = kz_json:get_value(CascadesQty, CurrentServicesJObj),
+    Diff = kz_json:diff(CQ, PQ),
+    case kz_json:is_empty(Diff) of
+        'true' -> 'false';
+        _ -> 'true'
+    end.
+
+%%------------------------------------------------------------------------------
+%% @doc extracts diff of services and returns audit changes JObj
+%% @end
+%%------------------------------------------------------------------------------
+-spec generate_diff_obj(kz_json:object(), kz_json:object()) -> kz_json:object().
+generate_diff_obj(CurrentServicesJObj, ProposedServicesJObj) ->
+    Key = [<<"quantities">>],
+    PQ = kz_json:get_value(Key, ProposedServicesJObj),
+    CQ = kz_json:get_value(Key, CurrentServicesJObj),
+    Diffs = kz_json:diff(CQ, PQ),
+    Keys = kz_json:get_keys(kz_json:flatten(Diffs)),
+    calculate_diffs({Keys, CQ, PQ}, []).
+
+%%------------------------------------------------------------------------------
+%% @doc generates the changes JObj for each diff key with changed and billable
+%% quantities
+%% @end
+%%------------------------------------------------------------------------------
+-spec calculate_diffs({kz_json:objects(), kz_json:object(), kz_json:object()}, kz_json:object()) -> kz_term:proplist().
+calculate_diffs({[], _, _}, Acc) -> Acc;
+calculate_diffs({[Key | Keys], CQ, PQ}, Acc) ->
+    Proposed = kz_json:get_integer_value(Key, PQ, 0),
+    Current = kz_json:get_integer_value(Key, CQ, 0),
+    [_Source, Category, Item] = Key,
+    Obj = [{<<"category">>, Category}
```

Comment:
Generally a list of proplists is...odd. Should this be `[kz_json:from_list(Obj) | Acc]` and the return be `kz_json:objects()`?